include/ck/tensor_operation/gpu/device/device_layernorm.hpp

```diff
@@ -106,47 +142,49 @@ struct DeviceLayernorm : public BaseOperator
-                                                                        AccElementwiseOperation,
-                                                                        GridDesc_M_K,
-                                                                        GridDesc_K,
-                                                                        BlockSize,
-                                                                        MThreadClusterSize,
-                                                                        KThreadClusterSize,
-                                                                        MThreadSliceSize,
-                                                                        KThreadSliceSize,
-                                                                        XYSrcVectorDim,
-                                                                        XSrcVectorSize,
-                                                                        GammaSrcVectorSize,
-                                                                        BetaSrcVectorSize,
-                                                                        XYSrcVectorDim,
-                                                                        YDstVectorSize,
-                                                                        true>;
+    using GridwiseReduceLayernormGeneric =
+        GridwiseLayernormWelfordVariance_mk_to_mk<XDataType,
+                                                  GammaDataType,
+                                                  BetaDataType,
+                                                  YDataType,
+                                                  AccDataType,
+                                                  AccElementwiseOperation,
+                                                  GridDesc_M_K,
+                                                  GridDesc_K,
+                                                  BlockSize,
+                                                  MThreadClusterSize,
+                                                  KThreadClusterSize,
+                                                  MThreadSliceSize,
+                                                  KThreadSliceSize,
+                                                  XYSrcVectorDim,
+                                                  XSrcVectorSize,
+                                                  GammaSrcVectorSize,
+                                                  BetaSrcVectorSize,
+                                                  XYSrcVectorDim,
+                                                  YDstVectorSize,
+                                                  false>;
+
+    using GridwiseReduceLayernormSweepOnce =
+        GridwiseLayernormWelfordVariance_mk_to_mk<XDataType,
+                                                  GammaDataType,
+                                                  BetaDataType,
```

Comment:
Since both `gamma` and `beta` are model parameters, no need to use separate types for them

Reply:
Distribution of gamma and beta could be different.
Consider the case of mix precision, type of each model parameter can be different.

Reply:
You can keep them different